pilot/pkg/bootstrap/server.go

```diff
@@ -198,6 +198,9 @@ func NewServer(args PilotArgs) (*Server, error) {
 	if args.Namespace == "" {
 		args.Namespace = os.Getenv("POD_NAMESPACE")
 	}
+	if args.Config.ClusterRegistriesNamespace == "" {
+		args.Config.ClusterRegistriesNamespace = args.Namespace
```

Comment:
hm, what if pilot runs out of cluster mode and namespace parameter is not specified?

Reply:
in that case `args.Namespace` will be the pilot pod's namespace (`os.Getenv("POD_NAMESPACE")`).  See the code section right above this one.

Reply:
@tiswanso that was exactly why I asked,  if pilot runs `OUT of cluster` then line 199 will return `nil` as this variable is not defined.

Reply:
@sbezverk -- sorry, I misunderstood.  hmm... in that case perhaps we shouldn't start the secret controller since there's no args to direct us to watch something.  OR alternatively, we could revert to `istio-system` if `args.Namespace == nil`.

Reply:
I think having a `istio-system` namespace as a last resort is fine.